sounddevice.py

```diff
@@ -2349,6 +2399,157 @@ def __init__(self, channel_selectors):
 
 
+class MacCoreSettings(object):
+
+    def __init__(self, channel_selectors=None, change_dev_params=False, allow_rate_conv=True, rate_conv_quality='Max'):
+        """Mac Core Audio-specific input/output settings.
+
+        Objects of this class can be used as *extra_settings* argument
+        to `Stream()` (and variants) or as `default.extra_settings`.
+
+        Parameters
+        ----------
+        channel_selectors : iterable of int, optional
+            Support for opening only specific channels of a MacCore
+            device.  *channel_selectors* is a list of integers
+            specifying the (zero-based) channel numbers to use.  Core
+            Audio refers to this as a Channel Map, and this class
+            provides some minor translation to be API compatible with
+            AsioSettings.
+        change_dev_params : bool, optional
+            If ``True``, allows PortAudio to change things like the
+            device's frame size, which allows for much lower latency,
+            but might disrupt the device if other programs are using it,
+            even when you are just Querying the device.  ``False`` is
+            the default.
+        allow_rate_conv : bool, optional
+            In combination with the above flag, ``True`` causes the
+            stream opening to fail, unless the exact sample rates are
+            supported by the device.
+        rate_conv_quality : string, optional
+            This sets MacCore's sample rate conversion quality.
+            Possible values are: ``'Min'``, ``'Low'``, ``'Medium'``,
+            ``'High'``, or ``'Max'``.  ``'Max'`` is the default.
+
+        Examples
+        --------
+        Setting output channels when calling `play()`: (Output to 'Phones 9' and 'Phones 10' on RME Fireface UFX+)
+
+        >>> import sounddevice as sd
+        >>> maccore_out = sd.MacCoreSettings(channel_selectors=[8, 9])
+        >>> sd.play(..., extra_settings=maccore_out)
+
+        Setting default output channels:
+
+        >>> sd.default.extra_settings = maccore_out
+        >>> sd.play(...)
+
+        Setting input channels as well:  (Input from 'Mic 9' and 'Mic 10' on RME Fireface UFX+)
+
+        >>> maccore_in = sd.MacCoreSettings(channel_selectors=[8, 9])
+        >>> sd.default.extra_settings = maccore_in, maccore_out
+        >>> sd.playrec(..., channels=1, ...)
+
+        """
+        # Check inputs:
+        conversion_dict = {
+            'Min':    _lib.paMacCoreConversionQualityMin   ,
+            'Low':    _lib.paMacCoreConversionQualityLow   ,
+            'Medium': _lib.paMacCoreConversionQualityMedium,
+            'High':   _lib.paMacCoreConversionQualityHigh  ,
+            'Max':    _lib.paMacCoreConversionQualityMax   ,
+            }
+        if rate_conv_quality not in conversion_dict:
+            raise ValueError('rate_conv_quality must be one of {!r}'.format(list(conversion_dict.keys())))
+
+        if channel_selectors is not None:
```

Comment:
AsioSettings' input verification is so loose that I question the value of having it at all:
```
if isinstance(channel_selectors, int):
    raise TypeError('channel_selectors must be a list or tuple')
```

For example, if someone passes in some other invalid data type such as a string, dict, OrderedDict, etc..., then the informative TypeError would never be raised, and error detection would fall to whatever ```_ffi.new('int[]', channel_selectors)``` will throw.

Is there some performance concern for the additional input type checking?  If so, I could wrap it in an ```if __debug__:``` check so that it only gets invoked as often as the other existing assert's...

On the whole, I'm fine with simplifying or stripping out this input checking.  Is there a different example that strikes a better balance between overkill and insufficient?